docs/getting_started/java/first_program_in_java/index.md

```diff
@@ -1,373 +1,757 @@
+Click on this to reveal the detail information about the failure call.
 
-### Review
+![ActivityTaskFailed event details](images/activity-task-failed-details.png)
 
-Answer the following questions to see if you remember some of the more important concepts from this tutorial:
+### Retry Logic
 
-<details>
-<summary>
+Without Temporal Orchestration, you must implement timeout and retry logic within the service code.
+It makes it your code hard to read with your recovery logic living right next to business logic.
+With Temporal, you can specify timeout configurations in the Workflow code as Activity options.
 
-**What are four of Temporal's value propositions that you learned about in this tutorial?**
+Temporal offers multiple ways to specify timeouts, including [Schedule-To-Start Timeout](https://docs.temporal.io/concepts/what-is-a-schedule-to-start-timeout), [Schedule-To-Close Timeout](https://docs.temporal.io/concepts/what-is-a-schedule-to-close-timeout), [Start-To-Close Timeout](https://docs.temporal.io/concepts/what-is-a-start-to-close-timeout), and [Heartbeat Timeout](https://docs.temporal.io/concepts/what-is-a-heartbeat-timeout).
+By default your code will be retried forever, unless a Schedule-To-Close Timeout or Start-To-Close Timeout is specified.
+This is not a good outcome.
+You want your code to complete (best outcome) or fail sensibly (an acceptable outcome), not to hang forever.
+Failing "sensibly" means that you can revive your failed Workflows by changing the services it relies on or re-establishing other ways that ensure Workflow success.
 
-</summary>
+### Increase the bug severity
 
-1. Temporal gives you full visibility in the state of your Workflow and code execution.
-2. Temporal maintains the state of your Workflow, even through server outages and errors.
-3. Temporal makes it easy to timeout and retry Activity code using options that exist outside of your business logic.
-4. Temporal enables you to perform "live debugging" of your business logic while the Workflow is running.
+Next, you'll make the Workflow fail harder.
+That is, you'll override some code that makes `deposit` continue to fail and prevent the timeout and recovery mechanisms from stepping in.
 
-</details>
+1. Quit the Worker if it is still running.
 
-<details>
-<summary>
+1. Make sure the `deposit` will still fail.
 
-**How do you pair up Workflow initiation with a Worker that executes it?**
+   ```java
+   boolean activityShouldSucceed = false;
+   ```
 
-</summary>
+1. Increase out the maximum number of attempts from 5 to a much bigger number:
 
-Use the same Task Queue.
+   ```
+   .setMaximumAttempts(5000)
+   ```
 
-</details>
+1. Increase the schedule-to-close timeout from 5 seconds to a much bigger number:
 
-<details>
-<summary>
+   ```
+   .setScheduleToCloseTimeout(Duration.ofSeconds(5000))
+   ```
 
-**What do you have to do if you make changes to Activity code for a Workflow that is running?**
+1. Compile and run the updated Worker
```

Comment:
I typically don't compile the individual pieces. I just `mvn clean compile` and compile the whole thing. Then use `mvn exc` to run.

Reply:
Good feedback.